src/Compilers/CSharp/Portable/Declarations/SingleTypeDeclaration.cs

```diff
@@ -237,6 +237,14 @@ public bool Equals(TypeDeclarationIdentity other)
                     return false;
                 }
 
+                if (!thisDecl.Location.SourceTree.Equals(otherDecl.Location.SourceTree)
```

Comment:
>Equals [](http://example.com/codeflow?start=50&length=6)

Consider using reference equality. It would match `isLeftAFileTypeInSeparateSourceTreeFromRight()`, and it would avoid unintended perf issues if a `SyntaxTree` implementation overrides `Equals()`.